enumerable_methods.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    instance = self
+    for i in 0...instance.length
+      yield(instance[i])
+    end
+  end
+
+  def my_each_with_index
+    instance = self
+    for i in 0...instance.length
+      yield(instance[i], i)
+    end
+  end
+
+  def my_select
+    instance = self
+    result = []
+    instance.my_each do |e|
+      result.push(e) if yield(e)
+    end
+    result
+  end
```

Comment:
* [ ] if no block is given (i.e check for `block_given?` at the beginning of the method) the method should return an Enumerator